### PR TITLE
Revert interface changes from "store unencrypted size in the unencrypted_size column"

### PR DESCRIPTION
**Pull request by icewind1991**
This reverts https://github.com/nextcloud/server/pull/31966 and uses the changes from https://github.com/nextcloud/server/pull/32708 instead which removes the significant interface changes at the cost of a "less nice" solution.

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by CarlSchwan** [comment]
now this will break all the app I changed a few days ago. For files_lock the solution was https://github.com/nextcloud/files_lock/pull/77

File: lib/private/Files/Cache/Propagator.php [hunk 1]
```diff
@@ -116,18 +117,18 @@ public function propagateChange($internalPath, $time, $sizeDifference = 0) {
...
 				// in case of encryption being enabled after some files are already uploaded, some entries will have an unencrypted_size of 0 and a non-zero size
+				$eq = $builder->expr()->eq('unencrypted_size', $builder->expr()->literal(0, IQueryBuilder::PARAM_INT));
+				$sizeColumn = $builder->getColumnName('size');
+				$unencryptedSizeColumn = $builder->getColumnName('unencrypted_size');
 				$builder->set('unencrypted_size', $builder->func()->greatest(
 					$builder->func()->add(
-						$builder->func()->case([
-							['when' => $builder->expr()->eq('unencrypted_size', $builder->expr()->literal(0, IQueryBuilder::PARAM_INT)), 'then' => 'size']
-						], 'unencrypted_size'),
+						new QueryFunction("CASE WHEN $eq THEN $unencryptedSizeColumn ELSE $sizeColumn END"),
```
Comment (nickvergessen) [inline lib/private/Files/Cache/Propagator.php:125]:
```suggestion
						$builder->createFunction("CASE WHEN $eq THEN $unencryptedSizeColumn ELSE $sizeColumn END"),
```
So you don't need to use private API directly

**Review by nickvergessen — APPROVED** [review]

**Comment by PVince81** [comment]
do we still need this or do we keep this change for 25 at least ?

**Comment by danxuliu** [comment]
Please check #33484 for a regression introduced in this pull request (second item in the list, storage stats size after deleting the last file in a folder is wrong).

**Comment by icewind1991** [comment]
@danxuliu thanks for making the test, regression should be fixed now

**Comment by danxuliu** [comment]
> @danxuliu thanks for making the test, regression should be fixed now

I have run the integration tests again rebased on the latest commit and that issue is indeed fixed. Thanks!

**Comment by icewind1991** [comment]
> do we still need this or do we keep this change for 25 at least ?

Imo it still makes sense to merge this to prevent needless churn for apps

**Comment by PVince81** [comment]
@icewind1991 I'm worried that apps might have already adjusted by now, especially if the interface change was already released recently or was it not ?

**Comment by icewind1991** [comment]
24 doesn't have the interface change.

For apps that added casts like https://github.com/nextcloud/files_lock/pull/77, it will still work with this PR applied, only if apps extended the querybuilder and overwrote existing method should they need to adjust for this PR

**Comment by skjnldsv** [comment]
This breaks master,
```json
{
  "reqId": "r9OmEUeVynCcFsLI8Lvu",
  "level": 3,
  "time": "2022-08-17T13:37:17+00:00",
  "remoteAddr": "172.18.0.1",
  "user": "admin",
  "app": "PHP",
  "method": "GET",
  "url": "/ocs/v2.php/apps/notifications/api/v2/notifications",
  "message": "Declaration of OC\\DB\\QueryBuilder\\ExpressionBuilder\\ExpressionBuilder::comparison($x, string $operator, $y, $type = null): string must be compatible with OCP\\DB\\QueryBuilder\\IExpressionBuilder::comparison($x, string $operator, $y, $type = null): OCP\\DB\\QueryBuilder\\IQueryFunction at /var/www/nextcloud/lib/private/DB/QueryBuilder/ExpressionBuilder/ExpressionBuilder.php#119",
  "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/94.0.4606.81 Safari/537.36",
  "version": "25.0.0.5",
  "data": {
    "app": "PHP"
  }
}
```

**Comment by skjnldsv** [comment]
Hum, weird, they look identical! :thinking:

**Comment by skjnldsv** [comment]
https://github.com/nextcloud/server/pull/31966#discussion_r884345139

**Pull request merged**